metapool/sample_sheet.py

```diff
@@ -1488,42 +1524,99 @@ def __init__(self, path=None):
             }
 
 
-def load_sample_sheet(sample_sheet_path):
-    # Load the sample-sheet using various KLSampleSheet children and return
-    # the first instance that produces a valid sample-sheet. We assume that
-    # because of specific SheetType and SheetVersion values, no one sample
-    # sheet can match more than one KLSampleSheet child.
+def _parse_header(fp):
+    df = pd.read_csv(fp, dtype="str", sep=",", header=None,
+                     names=range(100))
 
-    sheet = AbsQuantSampleSheetv10(sample_sheet_path)
-    if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
-        return sheet
+    # pandas will have trouble reading a sample-sheet if the csv has a
+    # variable number of columns. This occurs in legacy sheets when a user
+    # has introduced one too many ',' characters in a line.
+    #
+    # the solution is to fix the number of initial columns at a high enough
+    # value to include all columns, name them with integers, and later
+    # truncate all columns that are entirely empty.
+    df.dropna(how='all', axis=1, inplace=True)
 
-    sheet = AmpliconSampleSheet(sample_sheet_path)
-    if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
-        return sheet
+    # remove all whitespace rows (drop all rows that are entirely empty)
+    df.dropna(how='all', axis=0, inplace=True)
 
-    sheet = MetagenomicSampleSheetv102(sample_sheet_path)
-    if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
-        return sheet
+    # remove all comments rows, whether they are at the top of the file (no
+    # longer supported, technically), or not.
+    comment_rows = df.index[df[0].str.startswith("#")].tolist()
```

Comment:
Does the use of `comment` on parse with `read_csv` not work?

Reply:
I have to punt that one to @charles-cowart since this is his code (which is already in main).

Reply:
Ah I didn't know this feature existed, thanks! :) However, we don't have a convention in the lab that '#' can't be used. Hence I'm not sure that I would replace what's existing given that comment will ignore whatever's right of the '#' character, even if it's not at the beginning of the line.